app/cli/internal/action/artifact_upload.go

```diff
@@ -59,6 +59,11 @@ func (a *ArtifactUpload) Run(filePath string) (*CASArtifact, error) {
 		return nil, fmt.Errorf("retrieving file information: %w", err)
 	}
 
+	if info.Size() == 0 {
+		a.Logger.Info().Msg("the file is empty, nothing to upload")
```

Comment:
I have a context question, most likely for @migmartri, but anybody who knows the answer is welcome to reply ofc.

What is the context for these files?
Would someone find legit, in the generic scenario, to produce and upload an empty file?

Reply:
Interesting point.

I think you are right, it doesn't harm to upload empty files. In fact, artifact material types in attestations can be empty, but this code doesn't affect that (this is only for `chainloop artifact upload` cmd.) 

The change in `uploader.go` is in fact what fixes the original panic #430, so I think it's ok if we remove 

```go
if info.Size() == 0 {
		a.Logger.Info().Msg("the file is empty, nothing to upload")
...
```

and just keep the change at `internal/casclient/uploader.go`.

What do you think?

> Note: This implementation is based on my prescription offline, so any blame is on me :)

Reply:
Makes sense. I'll update the PR with your sugested changes. Thanks!

Reply:
Ok perfect. I'll request changes to remember not to merge this until it's fixed. 
Thanks.